weblate/glossary/tasks.py

```diff
@@ -40,6 +45,68 @@ def sync_glossary_languages(pk: int, component: Component | None = None) -> None
         component.create_translations_task()
 
 
+@app.task(trail=False, autoretry_for=(Project.DoesNotExist, WeblateLockTimeoutError))
+def cleanup_stale_glossaries(project: int | Project) -> None:
+    """
+    Delete stale glossaries.
+
+    A glossary translation is considered stale when it meets the following conditions:
+    - glossary.language is not used in any other non-glossary components
+    - glossary.language is different from glossary.component.source_language
+    - It has no translation
+
+    Stale glossary is not removed if:
+    - the component only has one glossary component
+    - if is managed outside weblate (i.e repo != 'local:')
+    """
+    if isinstance(project, int):
+        project = Project.objects.get(pk=project)
+
+    languages_in_non_glossary_components: set[int] = set(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=False
+        ).values_list("language_id", flat=True)
+    )
+
+    glossary_translations = prefetch_stats(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=True
+        )
+        .prefetch()
+        .exclude(language__id__in=languages_in_non_glossary_components)
+        .exclude(language=F("component__source_language"))
+    )
+
+    component_to_check = []
+
+    def can_delete(_glossary: Translation) -> bool:
+        """
+        Check if a glossary can be deleted.
+
+        It is possible to delete a glossary if:
+        - it has no translations
+        - it is not the only glossary in the project
+        - it is managed by Weblate (i.e. repo == 'local:')
+        """
+        return all(
+            [
+                _glossary.stats.translated == 0,
+                len(_glossary.component.project.glossaries) > 1,
```

Comment:
Hmm, this it probably too strict. We want to clean up the automatically created glossary in the end. Removing local blank glossaries will be likely good enough:

```suggestion
```